### DISCUSSION TITLE
Adding Google Analytics 4 to a Quire project

### DISCUSSION THREAD

**Discussion by mmpadilla15** [thegetty/quire#533] [Get Help] [answered]
I couldn't find this in the documentation, and I'm surprised this hasn't come up in the Help Discussion as I'm sure I'm not the only one who wants to add Google Analytics 4 to my Quire project.

I know how to set up a site on the Google side, but I'm not sure how to set it up on the Quire side. Can anyone advise?

**Comment by geealbers** [comment, marked as answer]
The Hugo version of Quire you're working on doesn't have support for GA4 (only the previous version) built in. I haven't tested this by you should be able put the code Google provides into your project's `themes/default/layouts/partials/analytics.html` file, replacing what's there currently. This will add their snippet to the `<head>` of the project.

They may also ask you to add something within the `<body>` of the project, and that could be added into `themes/default/layouts/_default/baseof.html`. You'll just want to make sure the snippet you add is on its own line(s) and that you don't inadvertently add , delete, or change anything else in that file.

**Reply by mmpadilla15** [reply]
@geealbers Google says to add the code to "every page of your website, immediately after the `<head>` element."

But if I add the code to the analytics.html file, do I need to add it to _ALL_ HTML files?

Also, am I replacing *everything* on that page, from `<script>` to `</script>`

**Reply by geealbers** [reply]
Yes, **everything** in that analytics.html file can be replaced, and yes, this will ensure it shows up on every page of your site. Good questions.

**Reply by mmpadilla15** [reply]
@geealbers I added the GA4 code to the analytics.html file last night. As of this morning, GA4 isn't pulling any data. But the last time I did this was on a Friday and by Monday data was pulling. I'll keep checking over the next few days and let you know if I see anything. Fingers crossed!

**Reply by mmpadilla15** [reply]
@geealbers Well, as of now, GA4 still is _not_ pulling data from the Quire site. This is what my analytics.html file looks like:
![Screen Shot 2022-10-13 at 2 48 35 PM](https://user-images.githubusercontent.com/74433007/195695479-c8b41575-6191-49cd-b525-a1a06bcc6e3e.png)

Any thoughts?

**Reply by geealbers** [reply]
Ah! I overlooked something, sorry about that.

In `themes/default/layouts/partials/head.html`, look for these lines near the bottom:

```
{{ if .Site.GoogleAnalytics}}
{{ partial "analytics.html" . }}
{{ end }}
```
And make them just:

```
{{ partial "analytics.html" . }}
```

**Reply by mmpadilla15** [reply]
@geealbers OMG ... it looks like it's working! I can see one user (me) when I look at realtime! Thank you so much!
